### PR TITLE
fix(keybinds): prevent default behavior for shortcuts,add call shortcuts

### PR DESCRIPTION
**Pull request by AlexZakablukov**
…cut, move call method to vuex

**What this PR does** 📖
prevend default behavior for shortcuts, add call shortcut for call active chat, move call method from toolbar to vuex

**Which issue(s) this PR fixes** 🔨
[AP-1020](https://satellite-im.atlassian.net/browse/AP-1020)

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

**Review by Jekrimo — APPROVED** [review]

**Review by phillsatellite — APPROVED** [review]

**Comment by phillsatellite** [comment]
Found one little bug as I finished up testing, if user sets any keybinding to "control + letter" it will not work. In screen recording I set settings to "control + a" and was unable to open settings using that keybindin

https://user-images.githubusercontent.com/93608357/158236351-5e395cf3-7807-4614-a1d3-2f2c60bcbe95.mov

g

**Review by stavares843 — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/29093946/158429202-e27142e3-5484-4265-9631-232b6dbac0b6.mov


What's the goal of this PR? I can't add any keybinds on this branch, I use a mac, and mac has control and seems control was removed from this PR to be a ctrl

I don't have the ctrl on my mac, only control

Works on dev branch

**Comment by molimauro** [comment]
Whats the "single key already exists error"? Other than that i think its okay

**Comment by AlexZakablukov** [comment]
> Whats the "single key already exists error"? Other than that i think its okay

It means you can use shortcut with two letters, for example shift + r + f, because mousetrap library that uses in this project does not suport it

**Comment by molimauro** [comment]
> > Whats the "single key already exists error"? Other than that i think its okay
> 
> It means you can not use shortcut with two letters, for example shift + r + f, because mousetrap library that uses in this project does not support it

Ok i think its ok then!

**Comment by phillsatellite** [comment]
Tested: Flow in screen recording

 - User tries to save Meta + C (copy on mac) gets error message 

- User edits keybinding and sets it to Meta + V (paste on mac) doesn't get error message and can save keybinding 

Should this issue be fixed on this pr? 

https://user-images.githubusercontent.com/93608357/158890152-126eebd1-cdcc-476b-b991-bbedfffa9ffb.mov

**Comment by AlexZakablukov** [comment]
> Tested: Flow in screen recording
> 
> * User tries to save Meta + C (copy on mac) gets error message
> * User edits keybinding and sets it to Meta + V (paste on mac) doesn't get error message and can save keybinding
> 
> Should this issue be fixed on this pr?
> 
>  meta.v.mov

I added meta + v for mac and ctrl + v for windows into blocked shortcuts, the list of blocked shortcuts you can see here https://github.com/Satellite-im/Core-PWA/blob/f1f3d0b396216e96a1a432913784fb4832790d57/utilities/HotkeyList.ts, (31- 112 lines for windows and 114 to 188 for mac), if it needs to add more tell me, please.

**Review by molimauro — APPROVED** [review]

File: components/views/navigation/toolbar/Toolbar.vue [hunk 1]
```diff
@@ -143,33 +143,7 @@ export default Vue.extend({
...
-      const peer = this.$WebRTC.getPeer(identifier)
-
-      try {
-        await peer?.call.createLocalTracks(kinds)
-        await peer?.call.start()
-      } catch (error) {
-        if (error instanceof Error) {
-          this.$toast.error(this.$t(error.message) as string)
-        }
-      }
```
Comment (stavares843) [inline components/views/navigation/toolbar/Toolbar.vue:172]: is this on purpose? doesn't seem related to the PR

**Review by stavares843 — CHANGES_REQUESTED** [review]
Captura de ecrã 2022-03-22, às 04 01 56

dev branch

when using control, appear control

Captura de ecrã 2022-03-22, às 04 02 44

this branch, when using control appears ctrl

im using a mac, should appear control instead of ctrl

File: components/views/navigation/toolbar/Toolbar.vue (hunk 1 above)
Comment (AlexZakablukov) [reply components/views/navigation/toolbar/Toolbar.vue:172]: I moved this logic to vuex to make it possible to use via keybinds

**Comment by AlexZakablukov** [comment]
> Captura de ecrã 2022-03-22, às 04 01 56
> 
> dev branch
> 
> when using control, appear control
> 
> Captura de ecrã 2022-03-22, às 04 02 44
> 
> this branch, when using control appears ctrl
> 
> im using a mac, should appear control instead of ctrl

try to use you keybind with control on dev branch - it does not work, lib used in the project does not support control, I will try  to fix it

File: utilities/Keybinds.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+export const keyboardRegex = /[a-z0-9\t\n /<>?;:"'`!@#$%^&*()[\]{}_+=|\\-]/gm
+
+const isMac = navigator.platform.toLowerCase().includes('mac')
```
Comment (molimauro) [inline utilities/Keybinds.ts:3]: can you use this please:
```
typeof navigator !== 'undefined' && /Mac|iPad|iPhone|iPod/.test(navigator.userAgent)
```

**Comment by molimauro** [comment]
The control/ctrl visualization hack is fine for now, later on we can change library if needed

File: utilities/Keybinds.ts (hunk 2 above)
Comment (AlexZakablukov) [reply utilities/Keybinds.ts]: fixed

**Review by molimauro — APPROVED** [review]

**Comment by WanderingHogan** [comment]
I fixed the merge conflict, removed the ^ from faker in package (and moved it to dev deps), adding typings for mousetrap

**Review by stavares843 — APPROVED** [review]

**Pull request merged**